src/components/atom/AppDigitsValidator.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useRef } from "react";
+import DigitsValidatorProps from "../../types/DigitsValidatorProps";
+export const AppDigitsValidator = ({
+                    label,
+                    digitsCount,
+                    value,
+                    onChange,
+                    hasError,
+                    error,
+                    }: DigitsValidatorProps) => {
+    
+    const digitsRef = useRef<HTMLInputElement[]>([]);
+
+
+    const onDigitsChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const {value: fieldValue} = e.target;
+        const isOnlyNumbers = /^\d+$/.test(fieldValue);   
+
+        let insertedValue = (e.nativeEvent as InputEvent).data || '';
+
+
+        if(!isOnlyNumbers) {
+            e.target.value = fieldValue.replace(/[^0-9]/g, '');
+
+            if(fieldValue === "") { // user pressed backspace
+                const previousInput = e.target.previousElementSibling as HTMLInputElement;
+                if (previousInput) {
+                    previousInput.focus();
+                }
+            }
+        } else {
+            if(fieldValue.length > 1) {
+                e.target.value = insertedValue; // to replace the old value with the new one
+            }
+
+            const nextInput = e.target.nextElementSibling as HTMLInputElement;
+            if (nextInput) {
+                nextInput.focus();
+            }
+        }
+
+
+        let finalValue = '';
+
+        for(let i = 0; i < digitsRef.current.length; i++) {
+            finalValue += digitsRef.current[i].value;
+        }
+
+
+        value = finalValue;
+        onChange(finalValue);
+    };
```

Comment:
It's working great!

However the reading of the code is hard

I think it would be great if you could move some logic to a hook, see for example `src/components/molecules/RegisterIDForm.tsx` is a component using a hook

Because when I go to a component, I mainly want to see the html part to see the structure of the component, I don't want to see validation logic

Then, try to move this logic to a hook, so in `AppDigitsValidator ` we get to see only a component using a hook, but not the detailed logic of the validation

Also, use more functions, they are free! For example divide `onDigitsChange` in smaller functions to get you code more readable